cloudpathlib/cloudpath.py

```diff
@@ -305,28 +305,57 @@ def exists(self) -> bool:
     def fspath(self) -> str:
         return self.__fspath__()
 
-    def glob(self, pattern: str) -> Iterable["CloudPath"]:
-        # strip cloud prefix from pattern if it is included
-        if pattern.startswith(self.cloud_prefix):
-            pattern = pattern[len(self.cloud_prefix) :]
+    def _glob_checks(self, pattern):
+        if self.is_file():
+            raise NotImplementedError("Cannot run glob with a file as the root.")
+
+        if ".." in pattern:
+            raise NotImplementedError("Relative paths with '..' not supported in glob patterns.")
```

Comment:
I think we should make a new class in [`cloudpathlib.exceptions`](https://github.com/drivendataorg/cloudpathlib/blob/master/cloudpathlib/exceptions.py) like

```python
class CloudPathNotImplementedError(CloudPathException, NotImplementedError):
    pass
```

and use that.